Data_Structures_Answers.md

```diff
@@ -2,17 +2,78 @@ Add your answers to the questions below.
 
 1. What is the runtime complexity of your ring buffer's `append` method?
 
+
+#  def append(self, item):
+
+#    try:
+#        self.storage[self.current] = item -- O(1)
+#        self.current += 1 -- O(1)
+#
+#    except:
+#      self.current = 0 -- O(1)
+#      self.append(item) -- O(1)
+
+The time complexity seems to simplify to O(1)
+
 2. What is the space complexity of your ring buffer's `append` function?
 
+#  def append(self, item):
+
+#    try:
+#        self.storage[self.current] = item -- O(1)
+#        self.current += 1 -- O(1)
+#
+#    except:
+#      self.current = 0 -- O(1)
+#      self.append(item) -- O(1)
+
+The space complexity seems to simplify to O(1)... this is because at any given function call it is primarily using memory for two items ... the storage variable and the current variable.
+
 3. What is the runtime complexity of your ring buffer's `get` method?
 
+
+#  def get(self):
+#    items = [item for item in self.storage if item != None] -- O(1)
+#    return items -- O(1)
+
+The time complexity seems to simplify to O(1) ... Because it includes an assignment, and a return, which both are constant runtime.
```

Comment:
So if you fix the Big O of the comprehension you will get that this is actually O(n) based on how you are doing it